YoutubeExplode/Internal/SegmentedHttpStream.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class SegmentedHttpStream : Stream
+    {
+        private readonly HttpClient _httpClient;
+
+        private const int MaxSegmentSize = 9_898_989;
+
+        private readonly string _url;
+
+        Stream _currentStream;
+
+        private long _position;
+
+        public SegmentedHttpStream(HttpClient httpClient, string url, long length)
+        {
+            _url = url;
+            _httpClient = httpClient;
+            Length = length;
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => true;
+
+        public override bool CanWrite => false;
+
+        public override long Length { get; }
+
+        public override long Position
+        {
+            get => _position;
+            set
```

Comment:
What will happen when a user sets `Position`? I'm guessing since the stream isn't re-queried, it's not going to work.

Reply:
You are completely right, my bad. Fixed.